### PR TITLE
fix: change default listen address from `localhost` back to `network` and show all interfaces the service is listening on

### LINKED ISSUE #3593: `steampipe service start` and `service status` are not listening on `network` by default
Also, the output of `service start` and `service status` are incorrect for `*` . It's showing `[*]` whereas it should be showing all interfaces.

This broke during the time we merged the #3200 pull request from @pdecat .

### PR DESCRIPTION
**Pull request by pdecat**
Try and fix #3593.

With this change:

```
# steampipe service start --database-listen "*"
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop

# steampipe service status
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop

```

```
# steampipe service start --database-listen network
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1 100.106.208.87 172.17.0.1 172.18.0.1 10.0.17.107]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop

# steampipe service status
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1 100.106.208.87 172.17.0.1 172.18.0.1 10.0.17.107]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop
```

### CODE REVIEW

File: pkg/db/db_local/running_info.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ func newRunningDBInstanceInfo(cmd *exec.Cmd, listenAddresses []string, port int,
 		StructVersion:   RunningDBStructVersion,
 	}
 
+	if len(dbState.ListenAddresses) == 1 && dbState.ListenAddresses[0] == "*" {
+		addrs, _ := utils.LocalAddresses()
+		dbState.ListenAddresses = append(dbState.ListenAddresses, addrs...)
```
Comment (pdecat) [inline pkg/db/db_local/running_info.go:44]: Taken from https://github.com/turbot/steampipe/pull/3200/files#diff-5cdc872d3225f3c40af70a48a8baf0c3cc5757339aaad866130f0b342e83bc00L44

Shouldn't the first value be dropped instead?

**Comment by pdecat** [comment]
With the second commit:

```
# steampipe service start
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1 100.106.208.87 172.17.0.1 172.18.0.1 10.0.17.107]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop

# steampipe service status
Steampipe service is running:

Database:

  Host(s):            [* 192.168.1.5 192.168.122.1 192.168.39.1 100.106.208.87 172.17.0.1 172.18.0.1 10.0.17.107]
  Port:               9193
  Database:           steampipe
  User:               steampipe
  Password:           ********* [use --show-password to reveal]
  Connection string:  postgres://steampipe@127.0.0.1:9193/steampipe

Managing the Steampipe service:

  # Get status of the service
  steampipe service status

  # View database password for connecting from another machine
  steampipe service status --show-password

  # Restart the service
  steampipe service restart

  # Stop the service
  steampipe service stop
```

**Comment by binaek** [comment]
Hi @pdecat . Thank you for your prompt response on the issue and raising this PR.

I am already looking at this in the `3593-steampipe-service-start-and-service-status-are-not-listening-on-network-by-default` branch.

**Comment by pdecat** [comment]
Oh, ok then. I'll just close this PR :)

**Pull request closed without merging**